### PR TITLE
decoder/vaapidecoder_h265: split annexb and AVCC in h265 decoder

### PR DESCRIPTION
**Pull request by popkartyeah**
add the nalff mode in vaapidecoder_h265.cpp to support decode the mp4/mkv file.

### CODE REVIEW

File: decoder/vaapidecoder_h265.cpp [hunk 1]
```diff
@@ -1088,6 +1096,55 @@ YamiStatus VaapiDecoderH265::decode(VideoDecodeBuffer* buffer)
...
+    }
+    if (buf[0] != 1) {
+        VideoDecodeBuffer buffer;
+        memset(&buffer, 0, sizeof(buffer));
+        buffer.data = buf;
+        buffer.size = bufSize;
+        return (decode(&buffer) == YAMI_SUCCESS);
+    }
+
+    if (bufSize < 7) {
```
Comment (mypopydev) [inline decoder/vaapidecoder_h265.cpp]: Its wrong for HEVC Avc buf size check

**Comment by mypopydev** [comment]
LGTM

File: decoder/vaapidecoder_h265.cpp [hunk 2]
```diff
@@ -1088,6 +1096,52 @@ YamiStatus VaapiDecoderH265::decode(VideoDecodeBuffer* buffer)
...
+    }
+    if (bufSize < 24) {
+        ERROR("invalid avcc record data");
+        return false;
+    }
+    NalUnit nalu;
+    const uint8_t* nalBuf;
+    int32_t i = 0, j, numNalu, nalBufSize;
+    nalBuf = &buf[21];
+    m_nalLengthSize = (*nalBuf & 0x03) + 1;
```
Comment (lizhong1008) [inline decoder/vaapidecoder_h265.cpp]: Don't change m_nalLengthSize if function decodeAvcRecordData() return false. Else common/NalReader.cpp will treat it as avcc format. Please refer  https://github.com/01org/libyami/blob/apache/decoder/vaapidecoder_h264.cpp#L1180

Comment (lizhong1008) [inline decoder/vaapidecoder_h265.cpp]: LGTM except  m_nalLengthSize

**Comment by lizhong1008** [comment]
+1

**Comment by mypopydev** [comment]
+1

**Comment by xuguangxin** [comment]
Hi @wudping 
Could you help to verify this? If you do not have the mp4 file. Please ask from @mypopydev and add to  our daily regression.

**Comment by wudping** [comment]
Hi Guangxin，
Ok, I will.

Best Regards,
Dongping

**Comment by xuguangxin** [comment]
Hi @wudping 
Is any update on this?

**Comment by wudping** [comment]
verify+1

**Comment by xuguangxin** [comment]
merged as e2bc2a652ac6484a9a11c7ea9baa8568767a3968.
thanks @popkartyeah  and @mypopydev

**Pull request closed without merging**